### PR TITLE
Add MultiBox API

### LINKED ISSUE #8399: Wrap Python API for MultiBox
related https://github.com/PaddlePaddle/Paddle/issues/7488

### PR DESCRIPTION
**Pull request by chengduoZH**
fix #8399

### CODE REVIEW

File: python/paddle/v2/fluid/nets.py [hunk 1]
```diff
@@ -107,6 +108,38 @@ def __extend_list__(obj):
     return pool_out
 
 
+def img_conv_with_bn(input,
```
Comment (qingqing01) [inline python/paddle/v2/fluid/nets.py]: Do not wrap this API in Fluid.

File: python/paddle/v2/fluid/layers/detection.py [hunk 2]
```diff
@@ -16,14 +16,18 @@
 """
 
 from ..layer_helper import LayerHelper
+from ..param_attr import ParamAttr
 from ..framework import Variable
+from ..nets import img_conv_with_bn
 from tensor import concat
 from ops import reshape
+from nn import transpose
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/detection.py]:
```python
import ops
import nn
```

下面调用：`ops.reshape`  `ops.transpose`

File: python/paddle/v2/fluid/layers/detection.py [hunk 3]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+
+    Generate prior boxes' location and confidence for SSD(Single
+    Shot MultiBox Detector)algorithm. The details of this algorithm,
+    please refer the section 2.1 of SSD paper (SSD: Single Shot
+    MultiBox Detector)<https://arxiv.org/abs/1512.02325>`_ .
+
+    Args:
+       inputs(list): The list of input Variables, the format
+            of all Variables is NCHW.
+       num_classes(int): The number of calss.
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: calss --> classes

File: python/paddle/v2/fluid/layers/detection.py [hunk 4]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+       min_sizes(list, optional, default=None): The length of
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: list --> list|tuple

Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: length --> number

File: python/paddle/v2/fluid/layers/detection.py [hunk 5]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+            min_size is used to compute the the number of prior box.
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: remove `the`

File: python/paddle/v2/fluid/layers/detection.py [hunk 6]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+            If the min_size is None, it will be computed according
+            to min_ratio and max_ratio.
+       max_sizes(list, optional, default=None): The length of max_size
+            is used to compute the the number of prior box.
+       min_ratio(int): If the min_sizes is None, min_ratio and min_ratio
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: min_ratio --> max_ratio ?

File: python/paddle/v2/fluid/layers/detection.py [hunk 7]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+            will be used to compute the min_sizes and max_sizes.
+       max_ratio(int): If the min_sizes is None, min_ratio and min_ratio
+            will be used to compute the min_sizes and max_sizes.
+       aspect_ratios(list): The number of the aspect ratios is used to
+            compute the number of prior box.
+       base_size(int): the base_size is used to get min_size
+            and max_size according to min_ratio and max_ratio.
+       flip(bool, optional, default=False): Whether to flip
+            aspect ratios.
+       name(str, optional, None): Name of the prior box layer.
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: (str|None)

File: python/paddle/v2/fluid/layers/detection.py [hunk 8]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+            is used to compute the the number of prior box.
+       min_ratio(int): If the min_sizes is None, min_ratio and min_ratio
+            will be used to compute the min_sizes and max_sizes.
+       max_ratio(int): If the min_sizes is None, min_ratio and min_ratio
+            will be used to compute the min_sizes and max_sizes.
+       aspect_ratios(list): The number of the aspect ratios is used to
+            compute the number of prior box.
+       base_size(int): the base_size is used to get min_size
+            and max_size according to min_ratio and max_ratio.
+       flip(bool, optional, default=False): Whether to flip
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: bool|None

File: python/paddle/v2/fluid/layers/detection.py [hunk 9]
```diff
@@ -322,7 +325,200 @@ def _reshape_with_axis_(input, axis=1):
...
+            input=input,
+            num_filters=conv_num_filter,
+            filter_size=conv_filter_size,
+            padding=conv_padding,
+            stride=conv_stride,
+            param_attr=param_attr,
+            act=conv_act,
+            use_cudnn=use_cudnn)
+
+        if conv_with_batchnorm:
```
Comment (pkuyym) [inline python/paddle/v2/fluid/layers/detection.py]: Please double check here.

**Pull request merged**